tests/fixtures/character-class.js

```diff
@@ -44,25 +44,33 @@ const characterClassFixtures = [
 	{
 		pattern: '[^K]', // LATIN CAPITAL LETTER K
 		flags: 'u',
-		expected: '(?:[\\0-JL-\\uD7FF\\uE000-\\uFFFF]|[\\uD800-\\uDBFF][\\uDC00-\\uDFFF]|[\\uD800-\\uDBFF](?![\\uDC00-\\uDFFF])|(?:[^\\uD800-\\uDBFF]|^)[\\uDC00-\\uDFFF])',
+		matches: ["k", "\u212a", "\u{12345}", "\uDAAA", "\uDDDD"],
+		nonMatches: ["K"],
+		expected: '(?:[\\0-JL-\\uFFFF]|[\\uD800-\\uDBFF][\\uDC00-\\uDFFF])',
```

Comment:
They are now much shorter and easier to reason about. I also added `matches` tests so that we are confident that transpiled result is correct.